stats/statsreporter.go

```diff
@@ -88,24 +91,32 @@ func Post(destination string, stats []Sample) error {
 	payload := map[string][]Sample{"metrics": stats}
 	data, err := json.Marshal(payload)
 	if err != nil {
-		plog.Warningf("Couldn't marshal stats: ", err)
+		plog.WithField("stats", stats).WithError(err).
+			Warn("Couldn't marshal stats")
 		return err
 	}
-	statsreq, err := http.NewRequest("POST", destination, bytes.NewBuffer(data))
+	statsReq, err := http.NewRequest("POST", destination, bytes.NewBuffer(data))
 	if err != nil {
-		plog.Warningf("Couldn't create stats request: ", err)
+		plog.WithFields(logrus.Fields{
+			"destination": destination,
+			"data":        data,
+		}).WithError(err).Warn("Couldn't create stats request")
```

Comment:
Don't log a warning if you're returning the error. The higher level will handle the error as well, which would mean double logging.